src/components/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,664 @@
+   */
+  getA11yStatusMessage?: (options: DownshiftA11yStatusMessageOptions<DropdownListItem>) => string
+
+  /**
+   * A function that creates custom accessability message for dropdown item selection.
+   * @param {DropdownListItem} item - Dropdown selected element.
+   */
+  getA11ySelectedMessage?: (item: DropdownListItem) => string
+
+  /** A function that creates custom accessability message for dropdown item removal.
+   * @param {DropdownListItem} item - Dropdown removed element.
+   */
+  getA11yRemovedMessage?: (item: DropdownListItem) => string
+
+  /** A function that creates custom aria label accessability message for the remove item button.
+   * @param {DropdownListItem} item - The active item to be removed.
+   */
+  getA11yRemoveItemMessage?: (item: DropdownListItem) => string
+
+  /** Array of props for generating dropdown items and selected item labels if multiple selection. */
+  items?: DropdownListItem[]
+
+  /**
+   * Function to be passed to create selected searchQuery from selected item. It will be displayed on selection in the
+   * edit text, for search, or on the button, for non-search. Multiple search will always clear searchQuery on selection.
+   */
+  itemToString?: (item: DropdownListItem) => string
+
+  /** A dropdown can perform a multiple selection. */
+  multiple?: boolean
+
+  /** A string to be displayed when dropdown does not have available items to show. */
+  noResultsMessage?: string
+
+  /**
+   * Called on deletion by backspace when dropdown is a multiple search.
+   *
+   * @param {SyntheticEvent} event - React's original SyntheticEvent.
+   * @param {object} data - All props and proposed value.
+   */
+  onBackspaceDelete?: ComponentEventHandler<DropdownProps>
+
+  /**
+   * Callback for change in dropdown active value(s).
+   * @param {DropdownListItem|DropdownListItem[]} value - Dropdown active value(s).
+   */
+  onDropdownChange?: (value: DropdownListItem | DropdownListItem[]) => any
+
+  /**
+   * Called on clicking the 'X' icon corresponding to an active value.
+   *
+   * @param {SyntheticEvent} event - React's original SyntheticEvent.
+   * @param {object} data - All props and proposed value.
+   */
+  onCloseIconClick?: ComponentEventHandler<DropdownProps>
+
+  /**
+   * Called on key down on the 'X' icon corresponding to an active value.
+   *
+   * @param {SyntheticEvent} event - React's original SyntheticEvent.
+   * @param {object} data - All props and proposed value.
+   */
+  onCloseIconKeyDown?: ComponentEventHandler<DropdownProps>
+
+  /**
+   * Called on container click. Especially useful for multiple search case, when user
+   * can click the container outside the edit text or active values.
+   *
+   * @param {SyntheticEvent} event - React's original SyntheticEvent.
+   * @param {object} data - All props and proposed value.
+   */
+  onContainerClick?: ComponentEventHandler<DropdownProps>
```

Comment:
unfortunately, I cannot, so it may be better not to expose this to the user. I will remove the prop, and keep the private handler, that will focus the input each time a click is received in the dropdown container.